src/test/java/com/exactpro/junit5/WithByteBufExtension.java

```diff
@@ -0,0 +1,97 @@
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import static org.junit.platform.commons.util.AnnotationUtils.isAnnotated;
+
+public class WithByteBufExtension implements TestTemplateInvocationContextProvider {
+
+    private static final String METHOD_CONTEXT_KEY = "context";
+
+    @Override
+    public boolean supportsTestTemplate(ExtensionContext context) {
+        if (!context.getTestMethod().isPresent()) {
+            return false;
+        }
+        Method testMethod = context.getTestMethod().get();
+        if (!isAnnotated(testMethod, WithByteBuf.class)) {
+            return false;
+        }
+        WithByteBufMethodContext methodContext = new WithByteBufMethodContext(testMethod);
+        Preconditions.condition(methodContext.hasPotentiallyValidSignature(testMethod.getParameters()),
+            () -> String.format(
+                "@WithByteBuf method [%s] declares invalid formal parameter: "
+                    + "argument count must be 1 "
+                    + "argument type must be Collection class.",
+                testMethod.toGenericString()));
+
+        getStore(context).put(METHOD_CONTEXT_KEY, methodContext);
+        return true;
+    }
+
+    @Override
+    public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(
+        ExtensionContext extensionContext) {
+        Method templateMethod = extensionContext.getRequiredTestMethod();
+        String displayName = extensionContext.getDisplayName();
+        WithByteBufMethodContext methodContext = getStore(extensionContext)
+            .get(METHOD_CONTEXT_KEY, WithByteBufMethodContext.class);
+        String argValue = templateMethod.getAnnotation(WithByteBuf.class).value();
+        checkArgumentFormat(argValue);
+        return generateBufferArguments(argValue)
+            .stream()
+            .map(arguments -> createInvocationContext(
+                WithBufNameFormatter.formatName(displayName, arguments), methodContext, arguments));
+    }
+
+    private void checkArgumentFormat(String hexString) throws IllegalArgumentFormatException {
+        try {
+            ByteBufUtils.fromHex(hexString);
+        } catch (Exception e) {
+            throw new IllegalArgumentFormatException("Illegal Argument Format. " +
+                "Need to pass 1 byte hexadecimal values separated by a single space. " +
+                "regex format \"([0-9|a-f]{2} )\"");
+        }
+    }
+
+    private ArrayList<ArrayList<ByteBuf>> generateBufferArguments(String hexString) {
+        ArrayList<ByteBuf> bufArray = new ArrayList<>();
+        ArrayList<ArrayList<ByteBuf>> argsArray = new ArrayList<>();
+        byte[] bytes = ByteBufUtils.bytesFromHex(hexString);
+
+        //add entire array as a first testcase
+        bufArray.add(ByteBufUtils.buffFromBytes(bytes));
+        argsArray.add(new ArrayList<>(bufArray));
+        bufArray.clear();
```

Comment:
You create clone of bufArray for every test execution. In this case it's easier to do `bufArray = new ArrayList<>()` instead of `bufArray.clear()`